checkAmount/checkAmount/CheckAmountСalculator.cs

```diff
@@ -0,0 +1,168 @@
+﻿// <copyright file="CheckAmountСalculator.cs" company="Kalinin Andrew">
+// Copyright (c) Kalinin Andrew. All rights reserved.
+// </copyright>
+
+namespace CheckAmount;
+
+using System.Security.Cryptography;
+using System.Text;
+
+/// <summary>
+/// Calculates the checksum (hash amount) for a given file or directory structure.
+/// </summary>
+public class CheckAmountСalculator
+{
+    private const int BufferSize = 4096;
+
+    /// <summary>
+    /// Initiates the single-threaded checksum calculation.
+    /// </summary>
+    /// <param name="path">The path to the file or directory.</param>
+    /// <returns>The MD5 hash.</returns>
+    /// <exception cref="FileNotFoundException">Thrown if the path does not exist.</exception>
+    public string Calculate(string path)
+    {
+        byte[] hashBytes;
+
+        if (File.Exists(path))
+        {
+            hashBytes = this.CalculateFileCheckAmount(path);
+        }
+        else if (Directory.Exists(path))
+        {
+            hashBytes = this.CalculateDirectoryCheckAmount(path);
+        }
+        else
+        {
+            throw new FileNotFoundException("Путь не найден");
+        }
+
+        return BitConverter.ToString(hashBytes).Replace("-", string.Empty).ToLower();
+    }
+
+    /// <summary>
+    /// Initiates the multi-threaded checksum calculation for directories.
+    /// </summary>
+    /// <param name="path">The path to the file or directory.</param>
+    /// <returns>The MD5 hash.</returns>
+    /// <exception cref="FileNotFoundException">Thrown if the path does not exist.</exception>
+    public string CalculateMultiThreaded(string path)
+    {
+        byte[] hashBytes;
+
+        if (File.Exists(path))
+        {
+            hashBytes = this.CalculateFileCheckAmount(path);
+        }
+        else if (Directory.Exists(path))
+        {
+            hashBytes = this.CalculateDirectoryAmountThreaded(path);
+        }
+        else
+        {
+            throw new FileNotFoundException("Путь не найден");
+        }
+
+        return BitConverter.ToString(hashBytes).Replace("-", string.Empty).ToLower();
+    }
+
+    private byte[] CalculateFileCheckAmount(string filePath)
+    {
+        string fileName = Path.GetFileName(filePath);
+        byte[] fileNameBytes = Encoding.UTF8.GetBytes(fileName);
+        using (MD5 md = MD5.Create())
+        using (FileStream fileStream = new FileStream(filePath, FileMode.Open, FileAccess.Read))
```

Comment:
```suggestion
        using (FileStream fileStream = new(filePath, FileMode.Open, FileAccess.Read))
```
Древний C#, значит генерено нейросетью :)